release/rancher/rancher.go

```diff
@@ -71,3 +83,64 @@ func rancherImages(imagesURL string) (string, error) {
 	}
 	return string(images), nil
 }
+
+func CheckRancherDockerImage(tag string) error {
+	ctx := context.Background()
+	org := "rancher"
+	repo := "rancher"
+	archs := []string{"amd64", "arm64", "s390x"}
+	exists, err := docker.CheckImageArchs(ctx, org, repo, tag, archs)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		return errors.New("image tag doesn't exist or doesn't have all architectures")
+	}
+	return nil
+}
+
+func CheckHelmChartVersion(tag string) error {
+	versions, err := rancherHelmChartVersions(rancherHelmRepositoryURL)
+	if err != nil {
+		return err
+	}
+	foundVersion := false
+	for _, version := range versions {
+		logrus.Debug("checking version " + version)
+		if tag == version {
+			logrus.Info("found chart for version " + version)
+			foundVersion = true
+			break
+		}
+	}
+	if !foundVersion {
+		return errors.New("failed to find chart for rancher app version " + tag)
+	}
+	return nil
+}
+
+func rancherHelmChartVersions(repoURL string) ([]string, error) {
+	httpClient := http.Client{Timeout: time.Second * 15}
+	logrus.Debug("downloading: " + repoURL)
+	resp, err := httpClient.Get(repoURL)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		return nil, errors.New("failed to download index.yaml file, expected status code 200, got: " + strconv.Itoa(resp.StatusCode))
+	}
+	index, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	var helmIndex HelmIndex
+	if err := yaml.Unmarshal(index, &helmIndex); err != nil {
```

Comment:
There seems to be a decoder that can be used on this so the entire resp body doesn't need to be read all at once.